packages/instruction-plans/src/transaction-plan-executor.ts

```diff
@@ -29,10 +31,14 @@ export type TransactionPlanExecutor<TContext extends TransactionPlanResultContex
     config?: { abortSignal?: AbortSignal },
 ) => Promise<TransactionPlanResult<TContext>>;
 
+type ExecuteResult<TContext extends TransactionPlanResultContext> = {
+    context?: TContext;
+} & ({ signature: Signature } | { transaction: Transaction });
```

Comment:
Nice! I love that the executor can keep returning `{ context, transaction }` but if they need something looser they can just return `{ context, signature }` instead. I also like that we don't allow them to offer both the `transaction` and the `signature` to avoid any mismatches.